### DISCUSSION TITLE
Mixed Issues

### DISCUSSION THREAD

**Discussion by trudyhood** [vpnhood/VpnHood#1] [General]
Mixed questions

**Comment by devsynck** [comment]
Please update 
**Copy the access key and execute ./VpnHoodServer.exe gen** to 
**Copy the access key and execute ./VpnHoodServer.exe** 

As user have already generated token in previous cmd.

**Reply by trudyhood** [reply]
Thank you for your contribution

**Comment by navidlukas** [comment]
I want to install VPNHood server on linux VPS (Ubuntu)? Kindly appreciated

**Reply by trudyhood** [reply]
I haven't checked it on Ununtu yet! But it suppose to work. Let us know if there is any problem.

**Reply by navidlukas** [reply]
Thanks for the reply, Im new to nuget, I did my best, but am stuck here
sudo apt-get install -y nuget
sudo apt-get install mono-complete
sudo nuget update -self
sudo nuget install vpnHood.server

install successful, what do I do from here? to generate access key.
And once generated, is there a way to see the access keys that have already been generated later? Are they stored somewhere?

**Reply by trudyhood** [reply]
There is no package for linux yet, but you can run it manually. see here:
https://github.com/vpnhood/VpnHood/wiki/How-to-run-VpnHood-Server-on-Linux!

We have a plan to build a package for it to make it easier.

**Reply by navidlukas** [reply]
Thank you! works great! :-]

**Comment by devsynck** [comment]
Hi @trudyhood , 

Just so you know, Great initiative. I was searching for such cross platform solution long time.  I'm new to to VPN concept and wondering does **VPS** knows what data going (in/out) through our VPN server ?

![TcpProxyDiagram](https://user-images.githubusercontent.com/71482238/106612449-1dd29200-658f-11eb-9a7f-0b88fb382721.png)




I'm planning to use VpnHood for personal use. Also if I were to provide VpnHood as a free-to-use service (like an app) for users, is there any legal requirement (license) or something ? Thanks !

**Reply by trudyhood** [reply]
@patil-vaibhav 
Feel free to open new discussion with new topic

**does VPS knows what data going (in/out) through our VPN server**
Please check this link:
https://github.com/vpnhood/VpnHood/wiki/Can-a-VPN-server-monitor-or-change-my-internet-traffic

**is there any legal requirement (license) or something**
There is no license required to use VpnHood. you can run your own servers and give it access to any users you want

**Comment by Viperlk** [comment]
Hi Trudyhood, guys first of all thank you for what you guys do.
I just finished installing vpnhood on my android cellphone. I see it connects to the public servers and there is space to add servers. How or where do one obtain keys for other servers and is it even necessary for me to load extra servers .
I'm new to VPN usage.

**Comment by trudyhood** [comment]
@Viperlk Hi and welcome; VpnHood is a software solution, and as you said, we provide some public servers for testing purposes only and perhaps do not work in some countries. Unfortunately, we don't offer any private servers at the moment. However, you can find an IT expert to create a private server that requires paying for that server. You may also find some third parties that run their own VpnHood servers so they can give them access keys. 
It is unlikely to find a private server in public; if so, it is not private anymore. So you need to install it on your own or at least know someone.

I know many people are connected via VpnHood from some countries with server internet filtering, like China, Russia, and Iran, so I hope you can find a private server too.

**Comment by Milad305** [comment]
Hello i had one server on my server farm .
When i added another server my access keys need to regenerate .
And i was had to send new access keys to my customers .
Is this happen every time if i add a server to my server farm?
Becuse its not intresting that to regenerate hole access keys every time that i want to add another server

**Reply by trudyhood** [reply]
Sorry I missed your question.
When you add a new server to your server farm, there is no need to regenerate tokens. However, it is important to ensure that you do not delete the server that has the PublicInToken access point. If you do so, VpnHood will still be able to locate other servers by using the first server access point.

**Comment by tribird0** [comment]
I have 2 servers in my server farm and when I add a new server (the server location is not the same),
Could not find new server location on user side; I had to send new access keys to my clients. If the server location is the same. The client side option is not on the new server.
How can I make the new server appear in the user's side every time I add a new server?

**Reply by trudyhood** [reply]
The new locations will be updated after the clients establish the connection. Make sure you have selected "Push token to clients" in the server farm.
![image](https://github.com/vpnhood/VpnHood/assets/72891124/62805acc-492e-4e5b-a19e-9e705bf7bdef)

Users may need to close and open the app again, which we will fix that in the future.

**Reply by ramdroid2015** [reply]
First of all special thanks to trudyhood and this great application then:

I guess we have to use "Farm Token url" to update user's tokens but I do not know how to set "Farm Token url"; maybe it could be a simple txt file on a server like http(s)://domain.com/token.txt 

If it's not correct would you please help me on that.

**Reply by trudyhood** [reply]
Thank you,
Yes, it is a simple Txt file in a static URL, usually in  CDN like Dropbox or your own private server
http(s)://domain.com/token.txt

Please note, 
If you set the URL after uploading the file, you will need to download the farm token and upload it again, as the URL in the token will have changed. In other words, sometimes you need to upload the token twice the first time.

It usually takes up to 5 minutes for the VpnHood Access Manager checker to detect the new file contents, as some CDNs cache the previous content.
